### DISCUSSION TITLE
New G7 fails to connect

### DISCUSSION THREAD

**Discussion by tcbh** [NightscoutFoundation/xDrip#3706] [Bug reports]
Dear all,

first of all a big thank you for all the development and support you are doing.

I have after an update of xdrip now with the second sensor the problem to establish the connection. For the last one I had to install the patched G7 app to establish the pairing and to take the connection over by xdrip then.

Now I have the same problem with the next one and I would like to have it working as before. 

I will send you to screenshots: The status page and the logs with OB1 extending logging.

From my perspective my issue is similar to the one with the linked t-slim pump (#3687), but as I have no pump paired with the sensor, I decided to open another discussion.

I hope you could help 🙂
Thanks, tcbh
![Screenshot_20241009-155313](https://github.com/user-attachments/assets/e5e1f893-d670-4f0f-a10d-ce9f590f7c6d)
![Screenshot_20241009-155332](https://github.com/user-attachments/assets/e6c80c77-64e5-4964-8153-eaf42cb4ddd5)

**Comment by Navid200** [comment]
You have inserted the sensor, right?

Is there any other app or device getting readings from it now?

Have disabled battery optimization for xDrip and Bluetooth?

**Reply by tcbh** [reply]
Thanks Navid200, yes all done and according to your settings guideline.
No ther is no other app installed (Dexcom app removed before inserting it).

The only issue I see, currently some hour are gone after inserting it. Does the sensor go into a "sleep mode" if no successful BLE connection can be established after a respective time?

Thanks and cheers

**Reply by Navid200** [reply]
No, it will look for connection as long as it has battery unless something else is already connected.

Why is there no sound on the phone?
You will not hear the pair request.

**Reply by tcbh** [reply]
I just had switched off the sound, but kept an eye on the notifications.

**Reply by Navid200** [reply]
Use a scanner to see if you can see the transmitter:
https://navid200.github.io/xDrip/docs/Bluetooth-Scanner.html

**Reply by tcbh** [reply]
Indeed the sensor was not visible in the App proposed by you.
But I guess the Dexcom really goes into deep sleep, if not paired after a dedicated time after having inserted it: I only hovered a few times over the sensor with the magnet of the injector and 15 seconds later the pairing request of xdrip appeared. 🤔
But obviously the sensor has completely restarted as the warming up phase is counting down in xdrip right now.

Let's see, if it really starts transmitting readings...

**Reply by tcbh** [reply]
Indeed the sensor delivers readings now. As starting time the pairing time seems to be considered. At least xdrip shows 10.5 days remaining days for the sensor. 

The only question is, if the sensor counts differently as I inserted it more than 10 hours before.

Really weird that story 🤔

**Reply by Navid200** [reply]
Interesting.  I had a sensor that never made any transmission so I called it in and got a replacement.
If I ever end up with one like that, I will give a try to bringing the magnet close to see if it triggers anything.

**Reply by tcbh** [reply]
I only tried that because I found a website mentioning the "wake-up" by magnet - not sure if it was for a G6 or a G7. But I don't find back the link...

**Reply by tcbh** [reply]
I found it, but unfortunately it's in German only, but it was a G7 with the same behaviour as mine:
https://www.reddit.com/r/dexcom/comments/1b0vdsf/fixing_a_g7_sensor_that_wont_pair_or_start/?tl=de

**Reply by Navid200** [reply]
Thanks so much

**Reply by jobermayr** [reply]
> I found it, but unfortunately it's in German only, but it was a G7 with the same behaviour as mine: https://www.reddit.com/r/dexcom/comments/1b0vdsf/fixing_a_g7_sensor_that_wont_pair_or_start/?tl=de

After 5 hours reinstalling xDrip and applying QR code, restarting the phone, searching whole internet and also leaving the house a few times, I see this issue and the mentioned method fixes also persistent `mismatch - wait` issue within 20 s.

Maybe xDrip should get some long descriptions within the app to help users: e. g. what specific "errors" mean and what to do to fix it without reading whole internet …

**Reply by Navid200** [reply]
The challenge would be to identify why xDrip is not getting a pair request.
There are different situations that could cause it.  From xDrip's perspective, they all look the same.  
What I mean is if xDrip is not getting an acknowledgment, it could be because the transmitter has not been activated as you experienced, but it could also be because G7 is already paired to something else.  What xDrip experiences in these cases is the same.

As far as having to read the entire internet, I have a guide that includes everything including this (under troubleshooting at the bottom):
https://navid200.github.io/xDrip/docs/Dexcom/G7.html

We are open to suggestions and we also appreciate help.
We don't have a shortage of ideas and directives.  What we have a shortage of are people who are willing to roll up their sleeves and work with us as contributors.

**Comment by jamorham** [comment]
If you are getting mismatch for a long time and no pair request you should do the following:
- Quadruple check that you have entered the 4 digit pairing code correctly in to xDrip
- Move away any old sensors out of bluetooth range that you have previously removed and may be laying around the house or move yourself away from where they might be.
